src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -0,0 +1,1115 @@
+      check_if_successful(hmi_apis::FunctionID::TTS_SetGlobalProperties)) {
+    smart_objects::SmartObjectSPtr msg_params =
+        MessageHelper::CreateTTSResetGlobalPropertiesRequest(result,
+                                                             application);
+    auto msg = MessageHelper::CreateMessageForHMI(
+        hmi_apis::messageType::request,
+        application_manager_.GetNextHMICorrelationID());
+    (*msg)[strings::params][strings::function_id] =
+        hmi_apis::FunctionID::TTS_SetGlobalProperties;
+
+    (*msg)[strings::msg_params] = *msg_params;
+    ProcessMessageToHMI(msg, false);
+  }
+}
+
+void ResumptionDataProcessor::AddSubscriptions(
+    ApplicationSharedPtr application,
+    const smart_objects::SmartObject& saved_app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  AddButtonsSubscriptions(application, saved_app);
+  AddPluginsSubscriptions(application, saved_app);
+}
+
+void ResumptionDataProcessor::AddButtonsSubscriptions(
+    ApplicationSharedPtr application,
+    const smart_objects::SmartObject& saved_app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (!saved_app.keyExists(strings::application_subscriptions)) {
+    LOG4CXX_DEBUG(logger_, "application_subscriptions section is not exists");
+    return;
+  }
+
+  const smart_objects::SmartObject& subscriptions =
+      saved_app[strings::application_subscriptions];
+
+  if (subscriptions.keyExists(strings::application_buttons)) {
+    const smart_objects::SmartObject& subscriptions_buttons =
+        subscriptions[strings::application_buttons];
+    mobile_apis::ButtonName::eType btn;
+    for (size_t i = 0; i < subscriptions_buttons.length(); ++i) {
+      btn = static_cast<mobile_apis::ButtonName::eType>(
+          (subscriptions_buttons[i]).asInt());
+      application->SubscribeToButton(btn);
+    }
+
+    ButtonSubscriptions button_subscriptions =
+        GetButtonSubscriptionsToResume(application);
+
+    ProcessMessagesToHMI(
+        MessageHelper::CreateOnButtonSubscriptionNotificationsForApp(
+            application, application_manager_, button_subscriptions));
+  }
+}
+
+ButtonSubscriptions ResumptionDataProcessor::GetButtonSubscriptionsToResume(
+    ApplicationSharedPtr application) const {
+  ButtonSubscriptions button_subscriptions =
+      application->SubscribedButtons().GetData();
+  auto it = button_subscriptions.find(mobile_apis::ButtonName::CUSTOM_BUTTON);
+
+  if (it != button_subscriptions.end()) {
+    button_subscriptions.erase(it);
+  }
+
+  return button_subscriptions;
+}
+
+void ResumptionDataProcessor::AddPluginsSubscriptions(
+    ApplicationSharedPtr application,
+    const smart_objects::SmartObject& saved_app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  for (auto& extension : application->Extensions()) {
+    extension->ProcessResumption(
+        saved_app,
+        [this](const int32_t app_id, const ResumptionRequest request) {
+          this->SubscribeToResponse(app_id, request);
+        });
```

Comment:
```suggestion
  const auto app_id = application->app_id();

  for (auto& extension : application->Extensions()) {
    extension->ProcessResumption(
        saved_app,
        [this, app_id](const ResumptionRequest request) {
          this->SubscribeToResponse(app_id, request);
        });
```

no need to have this passed in from another place

Reply:
@iCollin this code has been reworked completely in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/e601b42dccc527918547dd89edbefddad202ef9b